### DISCUSSION TITLE
__get_validators__ deprecation in v2 : custom types

### DISCUSSION THREAD

**Discussion by phanicode** [pydantic/pydantic#7438] [Question] [answered]
import numpy as np
from pydantic import BaseModel

class NumpyFloat64Type(BaseModel):
    @classmethod
    def __get_validators__(cls):
        yield cls.validate

    @classmethod
    def validate(cls, value) -> np.float64:
        print(f"Validating value: {value}")
        return np.float64(value)

class TestModel(BaseModel):
    value: NumpyFloat64Type

data = {'value': 42.0}

try:
    test_obj = TestModel(**data)
    print(test_obj)
except Exception as e:
    print(e)
This used to work in v1, and is now giving me a validation error in v2 , I have read the documentation and read [this discussion](https://github.com/pydantic/pydantic/discussions/6890) but I could not figure it out.I really need to enforce that they follow custom types. Thanks!

**Comment by adriangb** [comment, marked as answer]
See https://docs.pydantic.dev/latest/usage/types/custom/
